qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -289,3 +306,115 @@ def default_values(self) -> List[DefaultCalValue]:
+
+    def __init__(
+        self,
+        basis_gates: Optional[List[str]] = None,
+        default_values: Optional[Dict] = None,
+        target_pulses: bool = True,
+    ):
+        """Setup the library.
+
+        Args:
+            basis_gates: The basis gates to generate.
+            default_values: A dictionary to override library default parameter values.
+            target_pulses: If True (the default) then drives will be added to the target qubit
+                during the CR tones on the control qubit.
+        """
+        self._target_pulses = target_pulses
+        super().__init__(basis_gates, default_values)
+
+    @property
+    def __supported_gates__(self) -> Dict[str, int]:
+        """The supported gates of the library are two-qubit pulses for the ecr gate."""
+        return {"cr45p": 2, "cr45m": 2, "ecr": 2}
+
+    def default_values(self) -> List[DefaultCalValue]:
+        """The default values of the CR library."""
+        defaults = []
+        for name, schedule in self.items():
+            for param in schedule.parameters:
+                if "ch" not in param.name:
+                    value = self._default_values[param.name]
+                    defaults.append(DefaultCalValue(value, param.name, tuple(), name))
+
+        return defaults
+
+    def _build_schedules(self, basis_gates: Set[str]) -> Dict[str, ScheduleBlock]:
+        """Build the schedules of the CR library."""
+
+        schedules = dict()
+
+        tgt_amp = Parameter("tgt_amp")
+        sigma = Parameter("σ")
+        cr_amp = Parameter("amp")
+        cr_dur = Parameter("duration")
+        cr_rf = Parameter("risefall")
+        t_chan_idx = Parameter("ch1")
+        u_chan_idx = Parameter("ch0.1")
+        t_chan = pulse.DriveChannel(t_chan_idx)
+        u_chan = pulse.ControlChannel(u_chan_idx)
+
+        if "cr45p" in basis_gates:
+            with pulse.build(name="cr45p") as cr45p:
+                pulse.play(
+                    pulse.GaussianSquare(cr_dur, cr_amp, risefall_sigma_ratio=cr_rf, sigma=sigma),
+                    u_chan,
+                )
+
+                if self._target_pulses:
+                    pulse.play(
+                        pulse.GaussianSquare(
+                            cr_dur, tgt_amp, risefall_sigma_ratio=cr_rf, sigma=sigma
+                        ),
+                        t_chan,
+                    )
+
+            schedules["cr45p"] = cr45p
+
+        if "cr45m" in basis_gates:
+            with pulse.build(name="cr45m") as cr45m:
+                pulse.play(pulse.GaussianSquare(cr_dur, -cr_amp, width=cr_rf, sigma=sigma), u_chan)
+
+                if self._target_pulses:
+                    pulse.play(
+                        pulse.GaussianSquare(cr_dur, -tgt_amp, width=cr_rf, sigma=sigma), t_chan
+                    )
+
+            schedules["cr45m"] = cr45m
+
+        if "ecr" in basis_gates:
```

Comment:
Do you think you need validation for missing cr45p and m? If this is instantiated only with "ecr" you cannot get any subroutine to assign. On the other hand, avoiding the validation allows you to take cr45p and m from other library.

Reply:
I don't think we need to do that. The calibrations will complain if you try and reference a schedule that has not been registered yet. E.g.
```python
cals = Calibrations.from_backend(FakeBelemV2(), libraries=[
    FixedFrequencyTransmon(), EchoedCrossResonance(basis_gates=["ecr"])
])
```
If we do this validation in the library than we cannot register a cr45p from somewhere else as you correctly point.